File: php/features.php
<?php
declare(strict_types=1);

// Pokefact SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class PokefactFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new PokefactBaseFeature();
            case "test":
                return new PokefactTestFeature();
            default:
                return new PokefactBaseFeature();
        }
    }
}

File: py/features.py
# Pokefact SDK feature factory

from feature.base_feature import PokefactBaseFeature
from feature.test_feature import PokefactTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PokefactBaseFeature(),
        "test": lambda: PokefactTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
